src/Query/Admin/TelemetryQuery/DbalTelemetryQuery.php

```diff
@@ -0,0 +1,153 @@
+     */
+    public function organizations(int $limit = 100, int $offset = 0): array
+    {
+        return array_map(function (array $data): Organization {
+            return new Organization(
+                $data['id'],
+                $data['tokens'],
+                $data['has_anonymous_access'],
+                $data['members'],
+                $data['owners'],
+            );
+        }, $this->connection->fetchAll(
+            'SELECT
+                o.id,
+                COUNT(t.value) tokens,
+                o.has_anonymous_access,
+                COUNT(m.id) members,
+                COUNT(ow.id) owners
+            FROM "organization" o
+            LEFT JOIN "organization_token" t ON t.organization_id = o.id
+            LEFT JOIN "organization_member" m ON m.organization_id = o.id AND m.role = :role_member
+            LEFT JOIN "organization_member" ow ON ow.organization_id = o.id AND ow.role = :role_owner
+            GROUP BY o.id
+            LIMIT :limit OFFSET :offset',
+            [
+                ':role_member' => Member::ROLE_MEMBER,
+                ':role_owner' => Member::ROLE_OWNER,
+                ':limit' => $limit,
+                ':offset' => $offset,
+            ])
+        );
+    }
+
+    public function organizationsCount(): int
+    {
+        return (int) $this
+            ->connection
+            ->fetchColumn(
+                'SELECT COUNT(id) FROM "organization"'
+            );
+    }
+
+    /**
+     * @return Package[]
+     */
+    public function packages(string $organizationId, \DateTimeImmutable $till, int $limit = 100, int $offset = 0): array
+    {
+        return array_map(function (array $data): Package {
+            return new Package(
+                $data['type'],
+                $data['latest_release_date'] === null ? null : new \DateTimeImmutable($data['latest_release_date']),
+                $data['last_sync_at'] === null ? null : new \DateTimeImmutable($data['last_sync_at']),
+                $data['last_scan_date'] === null ? null : new \DateTimeImmutable($data['last_scan_date']),
+                $data['last_sync_error'] !== null,
+                $data['webhook_created_at'] !== null,
+                $data['last_scan_status'] ?? '',
+                $data['downloads'],
+                $data['webhooks'],
+            );
+        }, $this->connection->fetchAll(
+            'SELECT
+                p.type,
+                p.latest_release_date,
+                p.last_sync_at,
+                p.last_scan_date,
+                p.last_sync_error,
+                p.webhook_created_at,
+                p.last_scan_status,
+                COUNT(d.id) downloads,
+                COUNT(w.date) webhooks
+            FROM "organization_package" p
+            LEFT JOIN "organization_package_download" d ON d.package_id = p.id
+            LEFT JOIN "organization_package_webhook_request" w
+                ON w.package_id = p.id AND w.date::date <= :till
```

Comment:
please also check if a subquery wouldn't be better instead of such a join:
```sql
 p.last_scan_status,
(SELECT COUNT(d.id) FROM organization_package_download WHERE d.package_id = p.id AND d.date::date <= :till) AS downloads
```

Reply:
It looks like subqueries would perform better